File: programmers/가장_큰_수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        int len=numbers.length;
        String[] strnum = new String[len];
        for(int i=0;i<len;i++){
            strnum[i]= numbers[i]+"";
        }
        Arrays.sort(strnum,(o1,o2)->{
            return (o2+o1).compareTo(o1+o2);
        });
        if(strnum[0].startsWith("0")){
            answer="0";
        }
        else{
            for(int i=0;i<len;i++){
                answer+=strnum[i];
            }
        }
        return answer;
    }
}

File: programmers/level2/가장_큰_수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool comp(const string &a, const string &b) { 
    return a + b > b + a; // "3" + "30" > "30" + "3" 
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> num_to_string;
    for (int n : numbers)
         num_to_string.push_back(to_string(n));
    sort(num_to_string.begin(), num_to_string.end(), comp);
    
    if (num_to_string.front() == "0") {
        answer = "0";
    } else {
        for (string num : num_to_string)
            answer += num;
    }
    return answer;
}